include/vast/Conversion/ABI/AggregateTypes.hpp

```diff
@@ -62,6 +62,33 @@ namespace vast::conv::abi
         {
             return contains_subtype< hl::RecordType >(type);
         }
+
+        // TODO(conv:abi): Figure out how to instead use some generic helper.
+        gap::generator< ll::StructGEPOp > field_ptrs(
+            operation root, auto loc, auto &bld
+        ) const {
+            auto module_op = root->getParentOfType< vast_module >();
+            VAST_ASSERT(module_op);
+            VAST_ASSERT(root->getNumResults() == 1);
+
+            auto trg_type = [&] {
+                auto root_type = root->getResultTypes()[0];
+                if (auto ptr_type = mlir::dyn_cast< hl::PointerType >(root_type))
+                    return ptr_type.getElementType();
+                return root_type;
+            }();
+
+            auto def = hl::definition_of(trg_type, module_op);
+            std::size_t idx = 0;
+            for (const auto &[name, type] : def.getFieldsInfo()) {
+                auto ptr_type = hl::PointerType::get(root->getContext(), type);
```

Comment:
Can not you create a type here that is not in DL?

Reply:
Technically yes, but since all pointers are the same size DL will work as long as one pointer is there. This need more general fix, will give it a look over all our passes once we are on llvm-18.